docs/csharp/language-reference/keywords/in-generic-modifier.md

```diff
@@ -18,7 +18,7 @@ For generic type parameters, the `in` keyword specifies that the type parameter
   
  Contravariance enables you to use a less derived type than that specified by the generic parameter. This allows for implicit conversion of classes that implement variant interfaces and implicit conversion of delegate types. Covariance and contravariance in generic type parameters are supported for reference types, but they are not supported for value types.  
   
- A type can be declared contravariant in a generic interface or delegate if it is used only as a type of method arguments and not used as a method return type. `Ref` and `out` parameters cannot be variant.  
+ A type can be declared contravariant in a generic interface or delegate if it is used only as a type of method arguments and not used as a method return type. `In`, `ref` and `out` parameters cannot be variant.  
```

Comment:
You may have found a C# language spec issue.  Historically, `out` and `ref` args must be invariant. (Which is a more clear way of saying they cannot be variant, which means they can't be covariant or contravariant.) However, it would seem that `in` arguments could be contravariant. It might not be super useful, because `in` arguments are typically structs.

Reply:
I reworded. I now say that in ref and out parameters must be invariant.